src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -229,130 +233,233 @@ public String getUserAnalysisOutputsPage(Model model) {
-					tree(submission, model);
-				} else if (analysisType.equals(BuiltInAnalysisTypes.SISTR_TYPING)) {
-					model.addAttribute("sistr", true);
-				} else if (analysisType.equals(BuiltInAnalysisTypes.BIO_HANSEL)) {
-					model.addAttribute("bio_hansel", true);
-				}
-			}
+		//List of hashmaps which store the sample info
+		ArrayList<HashMap<String, Object>> sampleList = new ArrayList<>();
+
+		for (SampleFiles sampleFile : sampleFiles) {
+			HashMap<String, Object> hashSamples = new HashMap<>();
+			hashSamples.put("sampleName", sampleFile.getSample()
+					.getSampleName());
+			hashSamples.put("sampleId", sampleFile.getSample()
+					.getId());
+			hashSamples.put("sequenceFilePairId", sampleFile.getSequenceFilePair().getId());
+			hashSamples.put("forward", sampleFile.getSequenceFilePair()
+					.getForwardSequenceFile());
+			hashSamples.put("reverse", sampleFile.getSequenceFilePair()
+					.getReverseSequenceFile());
+			sampleList.add(hashSamples);
+		}
 
-		} catch (IOException e) {
-			logger.error("Couldn't get preview for analysis", e);
+		Map<String, Object> inputFilesMap = new HashMap<>();
+		inputFilesMap.put("result", "success");
+		inputFilesMap.put("samples", sampleList);
+
+		if (referenceFile != null) {
+			inputFilesMap.put("referenceFile", referenceFile);
 		}
 
-		return viewName;
+		return inputFilesMap;
 	}
 
 	/**
-	 * Update an analysis name
+	 * Update an analysis name and/or priority
 	 *
-	 * @param submissionId ID of the submission to update
-	 * @param name         name to update the analysis to
-	 * @param priority     the priority to update the analysis to.  Note only admins will be allowed to update priority
-	 * @param model        model for view
-	 * @param locale       locale of the user
+	 * @param parameters parameters which include the submission id and the new name and/or priority
+	 * @param locale     User's locale
+	 * @param response   HTTP response object
 	 * @return redirect to the analysis page after update
 	 */
-	@RequestMapping(value = "/{submissionId}/edit", produces = MediaType.TEXT_HTML_VALUE)
-	public String editAnalysis(@PathVariable Long submissionId, @RequestParam String name,
-			@RequestParam(required = false, defaultValue = "") AnalysisSubmission.Priority priority, Model model,
-			Locale locale) {
-		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
-
-		submission.setName(name);
-
-		boolean error = false;
-
-		try {
-			analysisSubmissionService.update(submission);
-			// Setting the priority as a separate call as it's not allowed to be updated with the normal update
-			if (priority != null) {
-				analysisSubmissionService.updatePriority(submission, priority);
+	@RequestMapping(value = "/ajax/update-analysis", method = RequestMethod.PATCH)
+	public Map<String, String> ajaxUpdateSubmission(@RequestBody AnalysisSubmissionInfo parameters, Locale locale,
```

Comment:
Trying to get away from return Maps can you create a dto for this?

Reply:
Fixed in 2804c23baa10fdf170ebd8b07a191142d0927b12